### PR TITLE
backport of iox-#654, iox-#791, iox-#802 and iox-#808 to release branch

### LINKED ISSUE #654: Starting an application after RouDi was terminated floods the cmd line with error messages
## Required information

**Operating system:**
Ubuntu based

**Compiler version:**
all

**Observed result or behaviour:**
Wenn RouDi is terminated and therefore couldn't cleanup the unix domain socket and an application is started, the cmd line gets flooded with
```
../iceoryx/iceoryx_utils/source/posix_wrapper/unix_domain_socket.cpp:401 { iox::cxx::expected<iox::posix::IpcChannelError> iox::posix::UnixDomainSocket::initalizeSocket(iox::posix::IpcChannelMode) }  :::  [ 111 ]  Connection refused
No server for unix domain socket "/tmp/roudi"
```

**Expected result or behaviour:**
The application should gracefully handle the situation and wait till RouDi is started

**Conditions where it occurred / Performed steps:**
Open a terminal and start `iox-roudi` and run `kill -9 [iox-roudi PID]`. Now run `./iox-ex-subscriber` and the cmd line get flooded with error messages

### PR DESCRIPTION
**Pull request by elBoberido**
## Pre-Review Checklist for the PR Author

1. [x] Code follows the coding style of [CONTRIBUTING.md][contributing]
1. [x] Tests follow the [best practice for testing][testing]
1. [x] Branch follows the naming format (`iox-#123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit messages are signed (`git commit -s`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [x] All checks have passed (except `task-list-completed`)
1. [x] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[contributing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#coding-style
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/advanced/best-practice-for-testing.md

## Notes for Reviewer

Backport of #804 to release_1.0 branch

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
- [x] Public API changes are documented via doxygen
- [x] Copyright owner are updated in the changed files
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## References

- Closes #654

### CODE REVIEW

**Comment by elBoberido** [comment]
@dkroenke I think we need to port #802 to the release branch

**Comment by dkroenke** [comment]
@elBoberido 
> I think we need to port #802 to the release branch

I cherry-picked the commit.

**Comment by elBoberido** [comment]
I guess I should also apply the fix for #791

**Comment by elBoberido** [comment]
@dkroenke shall I also backport the fix for #808 in this PR? I wanted to do a separate one but since the release branch is currently not building on some CI targets it might be better to do it here

**Comment by elfenpiff** [comment]
@elBoberido since you mentioned in issue #811 that the missing limits header is included in the correct cpptoml file via patch, could you please add the patch here. Or we do it with #812

File: iceoryx_dds/cmake/cpptoml/CMakeLists.txt [hunk 1]
```diff
@@ -63,7 +63,7 @@ endif()
 
 file(MAKE_DIRECTORY ${BUILD_DIR})
 
-execute_process(COMMAND ${CMAKE_COMMAND} -G "${CMAKE_GENERATOR}" "-DENABLE_LIBCXX=off" "-DCMAKE_INSTALL_PREFIX=${INSTALL_DIR}" "${TOOLCHAIN_FILE}" "${SOURCE_DIR}"
+execute_process(COMMAND ${CMAKE_COMMAND} -G "${CMAKE_GENERATOR}" "-DENABLE_LIBCXX=off" "-DCPPTOML_BUILD_EXAMPLES=off" "-DCMAKE_INSTALL_PREFIX=${INSTALL_DIR}" "${TOOLCHAIN_FILE}" "${SOURCE_DIR}"
```
Comment (mossmaurice) [inline iceoryx_dds/cmake/cpptoml/CMakeLists.txt:66]: Please update the copyright header

File: iceoryx_posh/cmake/cpptoml/CMakeLists.txt [hunk 2]
```diff
@@ -63,7 +63,7 @@ endif()
 
 file(MAKE_DIRECTORY ${BUILD_DIR})
 
-execute_process(COMMAND ${CMAKE_COMMAND} -G "${CMAKE_GENERATOR}" "-DENABLE_LIBCXX=off" "-DCMAKE_INSTALL_PREFIX=${INSTALL_DIR}" "${TOOLCHAIN_FILE}" "${SOURCE_DIR}"
+execute_process(COMMAND ${CMAKE_COMMAND} -G "${CMAKE_GENERATOR}" "-DENABLE_LIBCXX=off" "-DCPPTOML_BUILD_EXAMPLES=off" "-DCMAKE_INSTALL_PREFIX=${INSTALL_DIR}" "${TOOLCHAIN_FILE}" "${SOURCE_DIR}"
```
Comment (mossmaurice) [inline iceoryx_posh/cmake/cpptoml/CMakeLists.txt:66]: Please update the copyright header

**Review by mossmaurice — APPROVED** [review]

**Review by elfenpiff — APPROVED** [review]

**Pull request merged**